src/Entity/Activity/Registration.php

```diff
@@ -130,18 +138,30 @@ public function setOption(PriceOption $option): self
         return $this;
     }
 
-    public function getPerson(): ?LocalAccount
+    public function getPerson(): ?ContactInterface
     {
-        return $this->person;
+        return $this->person ?? $this->external;
     }
 
-    public function setPerson(?LocalAccount $person): self
+    public function setPerson(?ContactInterface $person): self
     {
-        $this->person = $person;
+        $this->person = null;
```

Comment:
Wouldn't this be neater?
```php
if ($person instanceof LocalAccount) {
    $this->external = null;
    $this->person = $person;
} elseif ($person instanceof ExternalRegistrant) {
    $this->person = null;
    $this->external = $person;
}
```

Reply:
I don't really feel like that's neater, no. The current setup structure better reflects what's happening: first clearing out any old values, then assigning the new value in the right place. This suggested structure makes that less clear.

Reply:
Although I understand what you're saying, I don't see why you need to set `this->person` to null before assigning it. That was the major point I tried to address with this edit.

Reply:
You're right, we don't need to set it, but whether we need to set a value or not is a premature optimization imo. The more important thing is to make clear what's happening here to any future developer. In my mind, what needs to happen here is two things: we need to (1) clear out the old value and (2) store the new value in the right place. When you group these steps together for each type of Contact (eg. within each `if` scope), a developer will need to analyse all the different contact here and make that generalization by themselves. Instead, by clearing all values first and then specializing the storing-part, the generalized structure is more evident automatically. Therefore, I still feel that the current structure is more clear, compared to this suggested edit.